Apim 3.18 audit braking changes (#767)
gravitee-io/gravitee-docs: pages/apim/3.x/installation-guide/upgrades/3.18.0/README.adoc

= Upgrade to 3.18.0

> == Breaking changes

Author: 🤔  to add or automatically added by the real release ?

=== Environment Audits

Before this version APIM management API will allow to get all audits of all environments and organization with this endpoint : `[GET] management/organizations/{orgId}/environments/{envId}/apis/{api}/audit`.  

Now this GET always return audit of the environment added in the url and no longer returns organization audits. 
To get organization audits you need to use the following endpoint : `[GET] management/organizations/{orgId}/audit`.

The query params `envLog:boolean` & `orgLog:boolean` are removed. 

> Now to get only environment or organization audit you have to use the query params `referenceType:string` with `ENVIRONMENT` or `ORGANIZATION`. You can also filter by environment id with `environment:string` query params.

Author: o nice. added

= Upgrade to 3.18.0

== Breaking changes

=== Environment Audits

> Endpoint :`[GET] management/organizations/{orgId}/environments/{envId}/apis/{api}/audit` (link:{{ '/apim/3.x/management-api/' | relative_url | append: current_version | append: '/index.html#operation/getApiAudits' }}[API audits^]).

Reviewer: ```suggestion
Endpoint:`[GET] management/organizations/{orgId}/environments/{envId}/apis/{api}/audit` (link:{{ '/apim/3.x/management-api/' | relative_url | append: current_version | append: '/index.html#operation/getApiAudits' }}[API audits^]).
```

> Before this version APIM management API will allow to get all audits of all environments and organization with this endpoint.

Reviewer: ```suggestion
Before this version, the Management API allowed getting all audits of all environments and organizations with this endpoint.
```

> Now this GET always return audit of the environment added in the url and no longer returns organization audits. 

Reviewer: ```suggestion
As of 3.18.0, a GET request will return the audits of the environment specified in the url and no longer return organization audits. 
```

> To get organization audits you need to use the following endpoint : `[GET] management/organizations/{orgId}/audit`.

Reviewer: ```suggestion
To get organization audits you need to use the following endpoint: `[GET] management/organizations/{orgId}/audit`.
```
